src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CryptoConfig.cs

```diff
@@ -439,5 +440,94 @@ public static string MapNameToOID(string name)
 
             return oidName;
         }
+
+        static public byte[] EncodeOID(string str)
+        {
+            if (str == null)
+                throw new ArgumentNullException(nameof(str));
+
+            char[] sepArray = { '.' }; // valid ASN.1 separators
+            string[] oidString = str.Split(sepArray);
+            uint[] oidNums = new uint[oidString.Length];
+            for (int i = 0; i < oidString.Length; i++)
+            {
+                oidNums[i] = (uint)Int32.Parse(oidString[i], CultureInfo.InvariantCulture);
+            }
+
+            // Allocate the array to receive encoded oidNums
+            byte[] encodedOidNums = new byte[oidNums.Length * 5]; // this is guaranteed to be longer than necessary
```

Comment:
Why not just pre-allocate the return `byte[]` by looping through oidNums to determine the exact size needed? This would avoid this intermediate array allocation.